.pre-commit-config.yaml

```diff
@@ -0,0 +1,53 @@
+repos:
+  - repo: local
+    hooks:
+      - id: extract-openapi
+        name: extract-openapi
+        entry: pipenv run extract-openapi
+        language: system
+        pass_filenames: false
+  - repo: local
+    hooks:
+      - id: generate-frontend-api-client
+        name: generate-frontend-api-client
+        entry: bash -c 'cd frontend && npm run codegen:v2:file'
```

Comment:
What's the trick that makes this command work without backend running?

Reply:
oh im answering myself...
Is it because the script `extract-openapi = "python backend/extract-openapi.py app.main:app"` can generate openapi.json without actually having a running background?
Does that mean I can also update the github action to eliminate all the steps of starting up all docker stacks?

Reply:
correct